src/arg_max_min.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './lib/tensor.js';
+import {reduceMax, reduceMin, selectValuesToReduce} from './reduce.js';
+import {squeeze} from './squeeze.js';
+
+/**
+ * Get the index location of the minimum or maxmium values of all the input values along the axes.
+ * @param {Tensor} input
+ * @param {Function} reduceFunc
+ * @param {MLArgMinMaxOptions} [options]
+ * @return {Tensor}
+ */
+export function argMaxMin(
+    input,
+    reduceFunc,
+    {
+      axes = null,
+      keepDimensions = false,
+      selectLastIndex = false,
+    } = {}) {
+  // If axes doesn't present (defaulting to null), all dimensions are reduced.
+  // See https://webmachinelearning.github.io/webnn/#dom-mlargminmaxoptions-axes.
+  const inpAxes = axes ?? new Array(input.rank).fill(0).map((_, i) => i);
```

Comment:
Regarding `axes ??`, I'm not that familiar with the falsesyness of JS evaluation. If you pass `axes = []`, will that correctly use the explicit empty `axes` (`output` shape is then the same size as the `input`, and there is a 1:1 mapping of output indices) rather than generate full reduction?

Reply:
> axes = []

It's so very special, it's neither **null** nor a sequence having values in the range [0, N-1]. Is there any model using this axes?

According to https://webmachinelearning.github.io/webnn/#dom-mlargminmaxoptions-axes:
```
axes, of type sequence<unsigned long>, defaulting to null
A sequence of [unsigned long](https://webidl.spec.whatwg.org/#idl-unsigned-long). The dimensions to reduce. The values in the sequence must be in the range [0, N-1] where N is the [rank](https://webmachinelearning.github.io/webnn/#mloperand-rank) of the input tensor. If not present, all dimensions are reduced.
```

Reply:
My argument is mainly driven by robustness plus consistency with similar behavior for the reduction operators when passed explicit `[]` axes. You probably won't find a model unless you look really hard, but some model generation parameters could programmatically produce it; and we see that in related reduction operators like ReduceSum that explicit empty axes `[]` was important enough that ONNX specifically added the `noop_with_empty_axes` attribute later https://onnx.ai/onnx/operators/onnx__ReduceSum.html.

Reply:
Last we also missed considering this **axes = []** cases in **reduceSum**. 
Since current Spec is lack of explanation for **axes = []**, I will open an issue for discussing it on WG.

Reply:
> I will open an issue for discussing it on WG.

Alrighty, and I will add my clear expectation of what should happen in the nop case 😉 (especially if WebNN is to support existing frameworks behavior where `[]` means nop).

Reply:
Link to https://github.com/webmachinelearning/webnn/issues/493

Reply:
That was fast! I tried ONNX and TF:

```
x = tf.constant([[0,100],[200,300]], dtype=tf.float32)
y = tf.math.reduce_sum(x, axis=[]) # https://www.tensorflow.org/api_docs/python/tf/math/reduce_sum
print("value:", y)
print("shape:", y.shape)

# Prints:
# value: tf.Tensor(
# [[  0. 100.]
#  [200. 300.]], shape=(2, 2), dtype=float32)
# shape: (2, 2)
```

(will add that to the issue...)

Reply:
Do you want to wait for the spec discussion or proceed and react later accordingly? (I don't mind either way, and you might not get a reply for the next 2 weeks...)

Reply:
Thanks. I will follow your guide to support **axes=[]** here.

Reply:
p.s. Appears that the Chromium implementation supports explicit `axes = []` gracefully:
https://chromium-review.googlesource.com/c/chromium/src/+/5125652